test/extended/images/prune.go

```diff
@@ -0,0 +1,261 @@
+		o.Expect(err).NotTo(o.HaveOccurred())
+		g.By(fmt.Sprintf("confirming storage size: sizeOfKeepImage=%d <= sizeAfterPrune=%d < beforePruneSize=%d", imgKeep.DockerImageMetadata.Size, confirmSize, keepSize))
+		o.Expect(confirmSize >= imgKeep.DockerImageMetadata.Size).To(o.BeTrue())
+		o.Expect(confirmSize < keepSize).To(o.BeTrue())
+		g.By(fmt.Sprintf("confirming pruned size: sizeOfPruneImage=%d <= (sizeAfterPrune=%d - sizeBeforePrune=%d)", imgPrune, keepSize, confirmSize))
+		o.Expect(imgPrune.DockerImageMetadata.Size <= keepSize-confirmSize).To(o.BeTrue())
+	}
+
+	g.Describe("of schema 1", func() {
+		g.JustBeforeEach(func() {
+			if *originalAcceptSchema2 {
+				g.By("ensure the registry does not accept schema 2")
+				err := ensureRegistryAcceptsSchema2(oc, false)
+				o.Expect(err).NotTo(o.HaveOccurred())
+			}
+		})
+
+		g.AfterEach(func() {
+			if *originalAcceptSchema2 {
+				err := ensureRegistryAcceptsSchema2(oc, true)
+				o.Expect(err).NotTo(o.HaveOccurred())
+			}
+		})
+
+		g.It("should prune old image", func() { testPruneImagesOfSchema(1) })
+	})
+
+	g.Describe("of schema 2", func() {
+		g.JustBeforeEach(func() {
+			if !*originalAcceptSchema2 {
+				g.By("ensure the registry accepts schema 2")
+				err := ensureRegistryAcceptsSchema2(oc, true)
+				o.Expect(err).NotTo(o.HaveOccurred())
+			}
+		})
+
+		g.AfterEach(func() {
+			if !*originalAcceptSchema2 {
+				err := ensureRegistryAcceptsSchema2(oc, false)
+				o.Expect(err).NotTo(o.HaveOccurred())
+			}
+		})
+
+		g.It("should prune old image with config", func() { testPruneImagesOfSchema(2) })
+	})
+})
+
+func getRegistryStorageSize(oc *exutil.CLI) (int64, error) {
+	ns := oc.Namespace()
+	defer oc.SetNamespace(ns)
+	out, err := oc.SetNamespace(kapi.NamespaceDefault).AsAdmin().Run("rsh").Args("dc/docker-registry", "du", "--bytes", "--summarize", "/registry/docker/registry").Output()
+	if err != nil {
+		return 0, err
+	}
+	m := regexp.MustCompile(`^\d+`).FindString(out)
+	if len(m) == 0 {
+		return 0, fmt.Errorf("failed to parse du output: %s", out)
+	}
+
+	size, err := strconv.ParseInt(m, 10, 64)
+	if err != nil {
+		return 0, fmt.Errorf("failed to parse du output: %s", m)
+	}
+
+	return size, nil
+}
+
+func doesRegistryAcceptSchema2(oc *exutil.CLI) (bool, error) {
+	ns := oc.Namespace()
+	defer oc.SetNamespace(ns)
+	env, err := oc.SetNamespace(kapi.NamespaceDefault).AsAdmin().Run("env").Args("dc/docker-registry", "--list").Output()
+	if err != nil {
+		return false, err
+	}
+
+	return strings.Contains(env, "REGISTRY_MIDDLEWARE_REPOSITORY_OPENSHIFT_ACCEPTSCHEMA2=true"), nil
+}
+
+// ensureRegistryAcceptsSchema2 checks whether the registry is configured to accept manifests V2 schema 2 or
+// not. If the result doesn't match given accept argument, registry's deployment config is updated accordingly
+// and the function blocks until the registry is re-deployed and ready for new requests.
+func ensureRegistryAcceptsSchema2(oc *exutil.CLI, accept bool) error {
```

Comment:
The only worry I have here, is that we might leave the registry in a state different than at the beginning, you are aware that this might influence other tests in the future? How about reverting it back to is original?